libs/eo/metering-points/shell/src/lib/eo-metering-points.store.ts

```diff
@@ -64,44 +56,71 @@ export class EoMeteringPointsStore extends ComponentStore<EoMeteringPointsState>
   }
 
   readonly loadingDone$ = this.select((state) => state.loadingDone);
-  readonly meteringPoints$ = this.select((state) => state.meteringPoints);
-  readonly error$ = this.select((state) => state.error);
-
-  readonly setLoadingDone = this.updater(
+  private readonly setLoadingDone = this.updater(
     (state, loadingDone: boolean): EoMeteringPointsState => ({
       ...state,
       loadingDone,
     })
   );
 
-  readonly setEnergySources = this.updater(
+  readonly meteringPoints$ = this.select((state) => state.meteringPoints);
+  private readonly setMeteringPoints = this.updater(
     (state, meteringPoints: EoMeteringPoint[]): EoMeteringPointsState => ({
       ...state,
       meteringPoints,
     })
   );
 
-  readonly setError = this.updater(
+  private readonly setCertificateContract = this.updater(
+    (state, contract: EoCertificateContract): EoMeteringPointsState => ({
+      ...state,
+      meteringPoints: state.meteringPoints.map((mp) => {
+        if (mp.gsrn === contract.gsrn) {
+          mp.contract = contract;
+        }
```

Comment:
At this point, this is fine as the contracts in the API only have a StartDate and not an EndDate, so please keep this as is. But in the future, EndDate will be added along with the possibility to cancel/stop a contract. This means, that it is possible to have multiple contracts per metering point.
FYI, the contracts returned by the API is only the contracts created by the signed-in user.

Reply:
Thanks :) Good to know for the future 😄